casanovo/denovo/dataloaders.py

```
@@ -342,6 +369,355 @@ def _make_dataset(
+        mzs,
+        intensities,
+        rt_offset,
+        **extra,
+    ):
+        """Append one scan's selected peaks onto the record for `key`,
+        creating it with any one-time metadata (extra) on first use."""
+        entry = prec_to_spec.setdefault(key, dict(extra))
+        entry.setdefault(scans_key, []).append(list(zip(mzs, intensities)))
+        entry.setdefault(rts_key, []).append(rt_offset)
+
+    def _extract_spectra(
```

Ruff (0.16.2)

[warning] Too many branches (14 > 12)

(PLR0912)